File: JavaScript/8js 3/twiceAsOld.js
/*
|--------------------------------------------------------------------------
| Twice as old
|--------------------------------------------------------------------------
|
*/

function twiceAsOld(d, s) {
   return (d - s) >= s ? (d - s) - s : ((d - s) - s) * -1
}


const twiceAsOld = (d, s) => Math.abs(d - s - s)


function twiceAsOld(dadYearsOld, sonYearsOld) {
   return Math.abs(2 * sonYearsOld - dadYearsOld);
}


function twiceAsOld(a, b) {
   return a > 2 * b ? a - 2 * b : 2 * b - a;
}


console.log(twiceAsOld(36, 7), 22);
console.log(twiceAsOld(55, 30), 5);
console.log(twiceAsOld(42, 21), 0);
console.log(twiceAsOld(44, 21), -2);
console.log(twiceAsOld(22, 1), 20);
console.log(twiceAsOld(29, 0), 29);

File: Python/8py 3/twice_as_old.py
"""
|--------------------------------------------------------------------------
| Twice as old
|--------------------------------------------------------------------------
|
"""

def twice_as_old(d, s):
   return abs(d - s - s)


twice_as_old = lambda d,s: ((d-s*2)**2)**0.5


def twice_as_old(d, s):
   return abs(d-2*s)


def twice_as_old(dad_years_old, son_years_old):
   return abs(2*son_years_old - dad_years_old)


print(twice_as_old(36,7) , 22)
print(twice_as_old(55,30) , 5)
print(twice_as_old(42,21) , 0)
print(twice_as_old(22,1) , 20)
print(twice_as_old(29,0) , 29)